.changelog/unreleased/improvements/3595-config-remove-genesishash.md

```diff
@@ -0,0 +1,2 @@
+- `[config]` Remove unused `GenesisHash` flag
+  ([\#3595](https://github.com/cometbft/cometbft/pull/3595))
```

Comment:
```suggestion
  ([\#3547](https://github.com/cometbft/cometbft/issues/3547)
```

Reply:
I will leave the PR number as it was also in our contributing guidelines.